### ISSUE TITLE
regression-new/array-haskell test segfaults on MacOS

### ISSUE THREAD

**Issue by Scott-Guest** [runtimeverification/k#3222]
Running `mvn verify` on MacOS reports a segfault in the array-haskell tests. Reproducing the test manually confirms this:
```
cd k-distribution/tests/regression-new/
make -C array-haskell
cd array-haskell
kompile --syntax-module TEST --no-exc-wrap --backend haskell test.k -d .
set -o pipefail; (cat 1.test.in 2>/dev/null || true) | krun 1.test --no-exc-wrap -d . | diff - 1.test.out
```

Note that if the test suit had additionally passed `--no-haskell-binary` to `kompile` above it would resolve the issue.

**Comment by ana-pantilie** [comment]
This is a known issue on macs, unfortunately there's no practical fix for it yet. @radumereuta could the test driver be modified in a way so that if the OS is a mac it always adds `--no-haskell-binary` to `kompile`?

**Comment by Baltoli** [comment]
> could the test driver be modified in a way so that if the OS is a mac it always adds --no-haskell-binary to kompile?

This is what the Nix build does, essentially. I think it should be reasonably straightforward to implement.

**Comment by Baltoli** [comment]
Fixed by #3230
